yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -241,45 +316,378 @@ class TQueueExportTask
     //! Original chunk specs fetched from master.
     std::vector<TChunkSpec> ChunkSpecs_;
     //! Pointers to chunk specs for chunks that are going to be exported within this iteration.
-    std::vector<const TChunkSpec*> ChunkSpecsToExport_;
-    //! Corresponds to the actual output table created.
-    TUserObject DestinationObject_;
-    TTransactionPtr UploadTransaction_;
-    //! Data statistics collected from attaching chunks.
-    TDataStatistics DataStatistics_;
+    TMap<ui64, std::vector<const TChunkSpec*>> ChunkSpecsToExportByUnixTs_;
 
     static constexpr TStringBuf ExportProgressAttributeName_ = "queue_static_export_progress";
     static constexpr TStringBuf ExportDestinationAttributeName_ = "queue_static_export_destination";
     static constexpr TStringBuf ExporterAttributeName_ = "queue_static_exporter";
 
-    ui64 GetMinFragmentUnixTs(TTimestamp timestamp)
+    struct TQueueExportSubtask final
     {
-        auto period = ExportConfig_.ExportPeriod.Seconds();
-        YT_VERIFY(period > 0);
+        const NNative::IClientPtr Client;
+        const NNative::IConnectionPtr Connection;
+
+        //! Corresponds to the queue being exported.
+        const TYPath Queue;
+        const TUserObject& QueueObject;
+        const TTableSchemaPtr& QueueSchema;
+        const TMasterTableSchemaId QueueSchemaId;
+
+        const TQueueStaticExportConfig ExportConfig;
+
+        //! The output table unix ts corresponding to the current export iteration.
+        const ui64 ExportFragmentUnixTs;
+        //! Instant of current export iteration.
+        const TInstant ExportInstant;
+
+        //! Pointers to chunk specs for chunks that are going to be exported within this iteration.
+        const std::vector<const TChunkSpec*>& ChunkSpecsToExport;
+
+        const ITransactionPtr ParentTransaction;
+
+        const TLogger Logger;
+
+        //! Options used for Cypress requests.
+        TTransactionalOptions Options;
+        //! Corresponds to the actual output table created.
+        TUserObject DestinationObject;
+        TTransactionPtr UploadTransaction;
+
+        //! Data statistics collected from attaching chunks.
+        TDataStatistics DataStatistics;
+
+        TQueueExportSubtask(
```

Comment:
`struct` in our code base usually don't have a constructor or public methods. It looks like a `class`.